SQL: Import Tableau connector

bpintea opened the pull request:
Initial Tableau connector import.

bpintea committed:
Import Tableau connector

Initial Tableau connector import.

bpintea committed:
Comment fixes

Remove (c) notices.

elasticmachine commented:
Pinging @elastic/es-ql (:Query Languages/SQL)

astefan reviewed (approved):
LGTM. I guess a follow up issue (and PR) is needed for adding Logstash data loading into Tableau.

bpintea committed:
Automate the TDVT test runs

Add script to automate the execution of test runs.
It still requires Tableau installed and licensed and the JDBC driver
pre-installed.

bpintea committed:
Add steps for loading test data with Logstash

Add the instructions and config file for loading the test data with
Logstash.

Also document running the tests automatically with the `tdvt_run.py`
tool.

Address review comment.

bpintea commented on x-pack/plugin/sql/connectors/tableau/README.md:
```diff
@@ -0,0 +1,20 @@
+ - Go to the [Connector Download](https://www.elastic.co/downloads/tableau-connector) page.
+ - Download the _.taco_ connector file.
+ - Move the _.taco_ file here:
+    - Windows: C:\Users[Windows User]\Documents\My Tableau Repository\Connectors
```
Fixed.

bpintea commented:
Thanks, @astefan.

> I guess a follow up issue (and PR) is needed for adding Logstash data loading into Tableau.

I've added a script to automise running the TDVT tools and added a paragraph on it, so I've added the steps for Logstash to this PR too. Let me know please if you'd like any changes to those or if it can go in as is.

bpintea commented:
@elasticmachine update branch

elasticmachine committed:
Merge branch 'master' into import_tableau_connector

bpintea commented on x-pack/plugin/sql/connectors/tableau/README.md:
```diff
@@ -0,0 +1,20 @@
+ - Go to the [Connector Download](https://www.elastic.co/downloads/tableau-connector) page.
+ - Download the _.taco_ connector file.
+ - Move the _.taco_ file here:
+    - Windows: C:\Users[Windows User]\Documents\My Tableau Repository\Connectors
```
```suggestion
    - Windows: C:\Users\[Windows User]\Documents\My Tableau Repository\Connectors
```

bpintea committed:
Address review comments

bpintea committed:
Update logstash.confn file

Fix logstash.conf version to add.

bpintea commented on x-pack/plugin/sql/connectors/tableau/tdvt/README.md:
```diff
@@ -0,0 +1,118 @@
+
+2. Adapt the [config file](https://www.elastic.co/guide/en/logstash/current/configuration-file-structure.html#configuration-file-structure) under the `logstash` folder, updating the <path>, <host> and <password> tags in it.
+
+3. Relaunch Logstash and wait until the files have been ingested. `calcs` index will need to have 17 documents and `staples` 54860.
```
```suggestion
3. Relaunch Logstash using the updated config file in previous step and wait until the files have been ingested. `calcs` index will need to have 17 documents and `staples` 54860.
```

bpintea committed:
Address review comments

bpintea commented on x-pack/plugin/sql/connectors/tableau/tdvt/README.md:
```diff
@@ -0,0 +1,117 @@
+
+Note that Logstash will add a few extra fields per row ("document" in Elasticsearch lingo); these shouldn't interfere with the testing, however.
+
+0. Download TDVT's data files [Calcs.csv](https://raw.githubusercontent.com/tableau/connector-plugin-sdk/tdvt-2.1.9/tests/datasets/TestV1/Calcs.csv) and [Staples_utf8.csv](https://raw.githubusercontent.com/tableau/connector-plugin-sdk/tdvt-2.1.9/tests/datasets/TestV1/Staples_utf8.csv) and place them into a directory that will be reachable by Logstash.
```
```suggestion
1. Download TDVT's data files [Calcs.csv](https://raw.githubusercontent.com/tableau/connector-plugin-sdk/tdvt-2.1.9/tests/datasets/TestV1/Calcs.csv) and [Staples_utf8.csv](https://raw.githubusercontent.com/tableau/connector-plugin-sdk/tdvt-2.1.9/tests/datasets/TestV1/Staples_utf8.csv) and place them into a directory that will be reachable by Logstash.
```

bpintea commented on x-pack/plugin/sql/connectors/tableau/tdvt/README.md:
```diff
@@ -0,0 +1,117 @@
+
+0. Download TDVT's data files [Calcs.csv](https://raw.githubusercontent.com/tableau/connector-plugin-sdk/tdvt-2.1.9/tests/datasets/TestV1/Calcs.csv) and [Staples_utf8.csv](https://raw.githubusercontent.com/tableau/connector-plugin-sdk/tdvt-2.1.9/tests/datasets/TestV1/Staples_utf8.csv) and place them into a directory that will be reachable by Logstash.
+
+1. Create the Elasticsearch indices `calcs` and `staples` and use for them the [mappings](https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-put-mapping.html) under these links: [calcs](https://github.com/elastic/elasticsearch-sql-odbc/blob/577cd2fa1ed257e42081a082682c8c089b179565/test/integration/data.py#L27) and [stapes](https://github.com/elastic/elasticsearch-sql-odbc/blo ...
```
```suggestion
2. Create the Elasticsearch indices `calcs` and `staples` and use for them the [mappings](https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-put-mapping.html) under these links: [calcs](https://github.com/elastic/elasticsearch-sql-odbc/blob/577cd2fa1ed257e42081a082682c8c089b179565/test/integration/data.py#L27) and [stapes](https://github.com/elastic/elasticsearch-sql-odbc/blob/577cd2fa1ed257e42081a082682c8c089b179565/test/integration/data.py#L87). Create an ingest [pipeline](https://www.elastic.co/guide/en/elasticsearch/reference/current/put-pipeline-api.html) called `calcs-pipeline` with the definition [here](https://github.com/elastic/elasticsearch-sql-odbc/blob/577cd2fa1ed257e42081a082682c8c089b179565/test/integration/data.py#L62).
```

bpintea commented on x-pack/plugin/sql/connectors/tableau/tdvt/README.md:
```diff
@@ -0,0 +1,117 @@
+
+1. Create the Elasticsearch indices `calcs` and `staples` and use for them the [mappings](https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-put-mapping.html) under these links: [calcs](https://github.com/elastic/elasticsearch-sql-odbc/blob/577cd2fa1ed257e42081a082682c8c089b179565/test/integration/data.py#L27) and [stapes](https://github.com/elastic/elasticsearch-sql-odbc/blo ...
+
+2. Adapt the [config file](https://www.elastic.co/guide/en/logstash/current/configuration-file-structure.html#configuration-file-structure) under the `logstash` folder, updating the <path>, <host> and <password> tags in it.
```
```suggestion
3. Adapt the [config file](https://www.elastic.co/guide/en/logstash/current/configuration-file-structure.html#configuration-file-structure) under the `logstash` folder, updating the <path>, <host> and <password> tags in it.
```

bpintea commented on x-pack/plugin/sql/connectors/tableau/tdvt/README.md:
```diff
@@ -0,0 +1,117 @@
+
+2. Adapt the [config file](https://www.elastic.co/guide/en/logstash/current/configuration-file-structure.html#configuration-file-structure) under the `logstash` folder, updating the <path>, <host> and <password> tags in it.
+
+3. Relaunch Logstash using the updated config file in previous step and wait until the files have been ingested. `calcs` index will need to have 17 documents and `staples` 54860.
```
```suggestion
4. Relaunch Logstash using the updated config file in previous step and wait until the files have been ingested. `calcs` index will need to have 17 documents and `staples` 54860.
```

bpintea commented on x-pack/plugin/sql/connectors/tableau/tdvt/README.md:
```diff
@@ -0,0 +1,117 @@
+
+### Automated
+
+0. Place Elasticsearch JDBC driver into [Tableau's driver folder](https://help.tableau.com/current/pro/desktop/en-us/examples_otherdatabases_jdbc.htm) under: `C:\Program Files\Tableau\Drivers`.
```
```suggestion
1. Place Elasticsearch JDBC driver into [Tableau's driver folder](https://help.tableau.com/current/pro/desktop/en-us/examples_otherdatabases_jdbc.htm) under: `C:\Program Files\Tableau\Drivers`.
```

bpintea commented on x-pack/plugin/sql/connectors/tableau/tdvt/README.md:
```diff
@@ -0,0 +1,117 @@
+
+0. Place Elasticsearch JDBC driver into [Tableau's driver folder](https://help.tableau.com/current/pro/desktop/en-us/examples_otherdatabases_jdbc.htm) under: `C:\Program Files\Tableau\Drivers`.
+
+1. Place the `.taco` file either in Tableau's dedicated connectors directory, `C:\Users\[Windows User]\Documents\My Tableau Repository\Connectors`, or a custom one ("<taco dir path>").
```
```suggestion
2. Place the `.taco` file either in Tableau's dedicated connectors directory, `C:\Users\[Windows User]\Documents\My Tableau Repository\Connectors`, or a custom one ("<taco dir path>").
```

bpintea commented on x-pack/plugin/sql/connectors/tableau/tdvt/README.md:
```diff
@@ -0,0 +1,117 @@
+
+1. Place the `.taco` file either in Tableau's dedicated connectors directory, `C:\Users\[Windows User]\Documents\My Tableau Repository\Connectors`, or a custom one ("<taco dir path>").
+
+2. Use the `tdvt_run.py` application, that clone the TDVT SDK repo, setup config files and launch the TDVT run:
```
```suggestion
3. Use the `tdvt_run.py` application, that clone the TDVT SDK repo, setup config files and launch the TDVT run:
```

bpintea commented on x-pack/plugin/sql/connectors/tableau/tdvt/README.md:
```diff
@@ -0,0 +1,117 @@
+
+Setting up the TDVT testing involves following the steps detailed in the [official documentation](https://tableau.github.io/connector-plugin-sdk/docs/tdvt). The "fragment" in parantheses reference the respective chapters in the documentation. It is recommended to execute each test run starting afresh.
+
+0. Same as in the automated testing.
```
```suggestion
1. Same as in the automated testing.
```

bpintea commented on x-pack/plugin/sql/connectors/tableau/tdvt/README.md:
```diff
@@ -0,0 +1,117 @@
+
+0. Same as in the automated testing.
+
+1. Create new Tableau data sources for the `calcs` and `Staple` tables (#`Test a new data source`), or, alternatively, use those available already in this repo.
```
```suggestion
2. Create new Tableau data sources for the `calcs` and `Staple` tables (#`Test a new data source`), or, alternatively, use those available already in this repo.
```

bpintea commented on x-pack/plugin/sql/connectors/tableau/tdvt/README.md:
```diff
@@ -0,0 +1,117 @@
+
+	Save the TDS files as `cast_calcs.elastic.tds` and `Staples.elastic.tds`
+
+2. Setup a TDVT "workspace" (#`Set up`), i.e. a directory containing the test files.
```
```suggestion
3. Setup a TDVT "workspace" (#`Set up`), i.e. a directory containing the test files.
```

bpintea commented on x-pack/plugin/sql/connectors/tableau/tdvt/README.md:
```diff
@@ -0,0 +1,117 @@
+	Copy/move the above saved `*.tds` files into the just created `tds` directory in the workspace.
+	Edit `config/tdvt/tdvt_override.ini` to update `TAB_CLI_EXE_X64` definition.
+
+3. Generate the tests by invoking TDVT as follows:
```
```suggestion
4. Generate the tests by invoking TDVT as follows:
```

bpintea commented on x-pack/plugin/sql/connectors/tableau/tdvt/README.md:
```diff
@@ -0,0 +1,117 @@
+
+	Edit the `elastic.ini` file in the `config` directory in the workspace and add the following line under the `[Datasource]` section: `CommandLineOverride = -DConnectPluginsPath=<path> -DDisableVerifyConnectorPluginSignature=true`, where `<path>` has the same value as in step "1.".
+
+4. Run the tests:
```
```suggestion
5. Run the tests:
```

bpintea committed:
Consistent numbering in the docs

Steps numbering brought inline with chapter numbering.

bpintea committed:
Address review comments

bpintea commented on x-pack/plugin/sql/connectors/tableau/tdvt/README.md:
```diff
@@ -0,0 +1,116 @@
+	When asked for a password, use the same value as in step "2." above when connecting to Elasticsearch.
+	For the "logical query config" use `simple_lower`.
+
+	Edit the `elastic.ini` file in the `config` directory in the workspace and add the following line under the `[Datasource]` section: `CommandLineOverride = -DConnectPluginsPath=<path> -DDisableVerifyConnectorPluginSignature=true`, where `<path>` has the same value as in step "1.".
```
```suggestion
	Edit the `elastic.ini` file in the `config` directory in the workspace and add the following line under the `[Datasource]` section: `CommandLineOverride = -DConnectPluginsPath=<path> -DDisableVerifyConnectorPluginSignature=true`, where `<path>` has the same value as in step "2.".
```

bpintea committed:
Address review comments

Update numbering references within text.

matriv commented on x-pack/plugin/sql/connectors/tableau/README.md:
```diff
@@ -0,0 +1,19 @@
+    -  macOS: /Users/[user]/Documents/My Tableau Repository/Connectors
+2. Elasticsearch JDBC Driver Installation:
+ - Go to the [Driver Download](https://www.elastic.co/downloads/jdbc-client) page.
+ - Download the Elasticsearch JDBC Driver _.jar_ file and move it into the following directory:
```
Should we mention something about appropriate versions based on the es node/cluster version?

bpintea replied:
That's a good point. I would however follow up on this:
* once the connector releasing is hooked into the RM (and we'll then easily be able to pair the connector with the driver, version-wise);
* with a doc PR stating our drivers' compatibility policy that we can refer to (from here, among other places).

bpintea committed:
Fix taco default dir definition

Remove hard-coded user with current login.

bpintea committed:
Merge branch 'import_tableau_connector' of github.com:bpintea/elasticsearch into import_tableau_connector

bpintea commented:
@elasticmachine update branch

elasticmachine committed:
Merge branch 'master' into import_tableau_connector

costin reviewed (commented):
Some of the files in this PR seem to have originated in the Tableau Connector SDK.
I'm not sure whether there's some git magic that can be used to download these directly from their original repo but assuming there's none or it is too complicated, it's important to clarify the situation and, if any of their resources are used, clearly indicate the source, copyright and license (links are enough).

bpintea committed:
Add Tableau's SDK MIT license

The x-pack/plugin/sql/connectors/tableau/connector/ files are modified
copies of ones provided by Tableau under:
 https://github.com/tableau/connector-plugin-sdk/tree/master/samples/plugins/postgres_jdbc
The x-pack/plugin/sql/connectors/tableau/tdvt/tds/ files are modified
files generated by the Tableau Desktop application.

bpintea commented:
@elasticmachine update branch

elasticmachine committed:
Merge branch 'master' into import_tableau_connector

costin reviewed (approved):
I wasn't clear in my original notes so here goes my LGTM
